kv/dist_sender.go

```diff
@@ -638,6 +638,16 @@ func (ds *DistSender) sendChunk(ctx context.Context, ba roachpb.BatchRequest) (*
 		var pErr *roachpb.Error
 		var finished bool
 		for r := retry.Start(ds.rpcRetryOptions); r.Next(); {
```

Comment:
we should be coping `rpcRetryOptions` and setting `Closer` to `ctx.Done()`; then you can check `ctx.Err()` outside the retry loop